PWGHF/D2H/Tasks/taskLb.cxx

```diff
@@ -34,33 +35,91 @@ using namespace o2::framework::expressions;
 
 void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
 {
-  ConfigParamSpec optionDoMC{"doMC", VariantType::Bool, true, {"Fill MC histograms."}};
+  ConfigParamSpec optionDoMC{"doMC", VariantType::Bool, false, {"Fill MC histograms."}};
   workflowOptions.push_back(optionDoMC);
 }
 
 #include "Framework/runDataProcessing.h"
 
 /// Λb0 analysis task
 struct HfTaskLb {
-  Configurable<int> selectionFlagLb{"selectionFlagLb", 1, "Selection Flag for Lb"};
+  Configurable<int> selectionFlagLb{"selectionFlagLb", 0, "Selection Flag for Lb"};
   Configurable<double> yCandGenMax{"yCandGenMax", 0.5, "max. gen particle rapidity"};
   Configurable<double> yCandRecoMax{"yCandRecoMax", 0.8, "max. cand. rapidity"};
+  Configurable<double> DCALengthParameter{"DCALengthParameter", 0.02, "decay length for DCA"};
+  Configurable<double> minLikelihoodRatio{"minLikelihoodRatio", 10., "min. likelihood ratio for combined DCAs"};
+  Configurable<double> minLikelihoodRatioLc{"minLikelihoodRatioLc", 10., "min. likelihood ratio for Lc cross check"};
   Configurable<std::vector<double>> binsPt{"binsPt", std::vector<double>{hf_cuts_lb_to_lc_pi::vecBinsPt}, "pT bin limits"};
 
   HfHelper hfHelper;
 
   Filter filterSelectCandidates = (aod::hf_sel_candidate_lb::isSelLbToLcPi >= selectionFlagLb);
+  float vtxCut = 10.;
+  Filter posZfilter = nabs(aod::collision::posZ) < vtxCut;
+
+  using TracksWExt = soa::Join<o2::aod::Tracks, o2::aod::TracksExtra, aod::TrackSelection, o2::aod::TrackSelectionExtension, aod::TracksPidPi, aod::PidTpcTofFullPi, aod::TracksPidKa, aod::PidTpcTofFullKa>;
+
+  double InvariantMass(auto track0, auto track1, double m0, double m1)
+  {
+    double px0 = track0.px();
+    double px1 = track1.px();
+    double pxSum = px0 + px1;
+    double py0 = track0.py();
+    double py1 = track1.py();
+    double pySum = py0 + py1;
+    double pz0 = track0.pz();
+    double pz1 = track1.pz();
+    double pzSum = pz0 + pz1;
+    double pSq0 = px0 * px0 + py0 * py0 + pz0 * pz0;
+    double pSq1 = px1 * px1 + py1 * py1 + pz1 * pz1;
+    double pSumSq = pxSum * pxSum + pySum * pySum + pzSum * pzSum;
+    double eSq0 = pSq0 + m0 * m0;
+    double eSq1 = pSq1 + m1 * m1;
+    double eTot = sqrt(eSq0) + sqrt(eSq1);
+    double eSq = eTot * eTot;
+    return sqrt(eSq - pSumSq);
+  }
```

Comment:
Implement a dynamic column in the candidate table and use `RecoDecay::m` for the actual calculation.